### PR TITLE
Adding lake with fish

### PR DESCRIPTION
**Pull request by MemphisMeng**

### CODE REVIEW

File: 01-python/outdoors/lake.py [hunk 1]
```diff
@@ -1,48 +1,65 @@
...
     """ 
     Draw a rectangular lake. 
   
     Parameters: 
         width (int): The width of the lake. 
         length (int): The length of the lake.    
     """
-    
+    r1 = int(random.random() * (width - 2) + 1)
+    r2 = int(random.random() * (length - 2) + 1)
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]: Please move this to after the parameter error handling (i.e. around line 14)

File: 01-python/outdoors/lake.py [hunk 2]
```diff
@@ -1,48 +1,65 @@
...
     for i in range(width):
         draw_vertical(False)
-        
+
         if i == 0 or i == width - 1:
             for _ in range(length):
                 draw_horizontal()
-        else:    
+        elif i == r1:
+            for _ in range(length):
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]:
```suggestion
            for j in range(length):
```

File: 01-python/outdoors/lake.py [hunk 3]
```diff
@@ -1,48 +1,65 @@
...
+                if _ == r2:
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]:
```suggestion
                if j == r2:
```

File: 01-python/outdoors/lake.py [hunk 4]
```diff
@@ -1,48 +1,65 @@
...
     """
     Draw a vertical fence.
     
     Parameter:
         newline (boolean): add a new line after drawing vertical component.
     """
-    if newline == False:
-        print("||", end = '')
+    if not newline:
+        print("||", end='')
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]:
```suggestion
        print("||", end = '')
```

File: 01-python/outdoors/lake.py [hunk 5]
```diff
@@ -1,48 +1,65 @@
...
         draw_vertical(True)
     return
 
 
 def draw_horizontal():
     """Draw a horizontal fence."""
-    print("=", end = '')
-    
-def draw_vertical(newline = False):
+    print("=", end='')
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]:
```suggestion
    print("=", end = '')
```

File: 01-python/outdoors/lake.py (hunk 1 above)
Comment (gallettilance) [reply 01-python/outdoors/lake.py]: And please change these variable names to something more descriptive (ex: `fishPositionX`, `fishPositionY`, `fishXCoordinate`, `fishYCoordinate`)

**Comment by gallettilance** [comment]
@MemphisMeng ping

**Comment by MemphisMeng** [comment]
Ok, I will fix them sure

**Comment by MemphisMeng** [comment]
Hi professor, I think I've squashed all the following commits to the initial one. But they are still left here. Do you know where can I find the problem??

**Comment by MemphisMeng** [comment]
I fixed them up! Finally! Thanks for your patience, Professor! Now you can safely review my work.

File: 01-python/outdoors/lake.py [hunk 6]
```diff
@@ -1,48 +1,65 @@
...
+        length (int): The length of the lake.
+    """
+
     if width < 0 or length < 0:
         raise ValueError("width and length must be greater than 0")
     elif width == 0 or length == 0:
         return
-        
+
+    fishCoordinateY = int(random.random() * (width - 2) + 1)
```
Comment (gallettilance) [inline 01-python/outdoors/lake.py]: this could be negative right?

File: 01-python/outdoors/lake.py (hunk 5 above)
Comment (gallettilance) [reply 01-python/outdoors/lake.py]: You forgot this :)

File: 01-python/outdoors/lake.py (hunk 4 above)
Comment (gallettilance) [reply 01-python/outdoors/lake.py]: This one too

**Comment by gallettilance** [comment]
@MemphisMeng ping

**Comment by MemphisMeng** [comment]
ping done!

**Comment by MemphisMeng** [comment]
OK

**Comment by MemphisMeng** [comment]
Professor, I just encountered some trouble here. I'll try to fix it by weekend.

**Comment by MemphisMeng** [comment]
Done, professor!

**Comment by gallettilance** [comment]
Great work!

**Pull request merged**